src/Dev/Tasks/TagsToShortcodeHelper.php

```diff
@@ -0,0 +1,290 @@
+                $content = str_replace($tag, $newTag, $content);
+            }
+        }
+
+        return $content;
+    }
+
+    /**
+     * Get all tags within some page content and return as array.
+     * @param $content
+     * @return array
+     */
+    private function getTagsInContent($content)
+    {
+        $resultTags = [];
+
+        preg_match_all('/<('.$this->validTagsPattern.').*?('.$this->validAttributesPattern.')\s*=.*?>/', $content, $matches, PREG_SET_ORDER);
+        if ($matches) {
+            foreach ($matches as $match) {
+                $resultTags []= $match[0];
+            }
+        }
+
+        return $resultTags;
+    }
+
+    /**
+     * @param string $tag
+     * @return array
+     */
+    private function getTagTuple(string $tag)
+    {
+        $pattern = sprintf(
+            '/.*<(?<tagType>%s).*(?<attribute>%s)="(?<src>[^"]*)"/i',
+            $this->validTagsPattern,
+            $this->validAttributesPattern
+        );
+        preg_match($pattern, $tag, $matches);
+        return $matches;
+    }
+
+    /**
+     * Gets the value of a given attribute from a tag.
+     * @param $tag
+     * @param $attribute
+     * @return null
+     */
+    private static function getAttribute($tag, $attribute)
+    {
+        $propertyValue = null;
+        $needle = $attribute . '="';
+        if (strpos($tag, $needle)) {
+            $propertyValue = explode('"', explode($needle, $tag)[1])[0];
+        }
+
+        return $propertyValue;
+    }
+
+    /**
+     * Extracts an array of attributes from a tag.
+     * @param string $tag
+     * @param string $tagType
+     * @param $id
+     * @param $newSrc
+     * @return array
+     */
+    private static function getAttributes(string $tag, string $tagType, $newSrc)
+    {
+        $attributes = [];
+
+        if ($tagType == 'img') {
+            $imgAttributes = [
+                'src' => $newSrc,
+            ];
+
+            if ($title = static::getAttribute($tag, 'title')) {
+                $imgAttributes['title'] = $title;
+            }
+            if ($width = static::getAttribute($tag, 'width')) {
+                $imgAttributes['width'] = $width;
+            }
+            if ($height = static::getAttribute($tag, 'height')) {
+                $imgAttributes['height'] = $height;
+            }
+            if ($class = static::getAttribute($tag, 'class')) {
+                $imgAttributes['class'] = $class;
+            }
+            if ($alt = static::getAttribute($tag, 'alt')) {
+                $imgAttributes['alt'] = $alt;
+            }
+
+            $attributes = array_merge($attributes, $imgAttributes);
+        }
+
+        return $attributes;
+    }
+
+    /**
+     * @param string $tagType
+     * @return string
+     */
+    private static function getShortCodeTypeFromTagType(string $tagType)
+    {
+        return static::VALID_TAGS[$tagType];
+    }
+
+    private function getParsedFileIDFromSrc(string $src)
```

Comment:
```suggestion
    private function getParsedFileIDFromSrc($src)
```

Add PHP doc.

Reply:
This method needs to be rethink a bit.

* What happens if the file hasn't been publish yet? You're not going to find it.
* You can probably get the expected file strategy directly from the FlysystemAssetStore
* URL always use forward slash.
* SoftResolve is meant to find files that use to exists, but have been archived. Is that really what you are going for here?